### PR TITLE
ICU-22687 Fix genren.pl not to skip C++ functions on Mac

### PR DESCRIPTION
**Pull request by richgillam**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22687
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [ ] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: icu4c/source/tools/genren/genren.pl [hunk 1]
```diff
@@ -154,7 +154,9 @@
             chop $qtype;
         } elsif ($mode =~ /Mach-O/) {
             ($address, $type, $_) = split(/ /);
-            if(/^_(.*)$/) {
+            if (/^(.*)\(/) {
+            	$_ = $1;
+            } elsif(/^_(.*)$/) {
```
Comment (markusicu) [inline icu4c/source/tools/genren/genren.pl:159]: I think a comment would be nice. I think this does
// Remove a leading underscore.

If you have any idea for why this is done here, maybe you could add that... maybe even an example?

File: icu4c/source/tools/genren/genren.pl [hunk 2]
```diff
@@ -154,7 +154,9 @@
             chop $qtype;
         } elsif ($mode =~ /Mach-O/) {
             ($address, $type, $_) = split(/ /);
-            if(/^_(.*)$/) {
+            if (/^(.*)\(/) {
```
Comment (markusicu) [inline icu4c/source/tools/genren/genren.pl:157]: Like below, a comment would be nice for what we do here and why. Maybe
// On macOS, C++ function symbols include their full signature, with (arguments in parens). Remove those.

**Review by pedberg-icu — APPROVED** [review]
Thanks for investigating and fixing this! Should have been done long ago 🙃

File: icu4c/source/tools/genren/genren.pl (hunk 1 above)
Comment (richgillam) [reply icu4c/source/tools/genren/genren.pl:159]: Good idea.  Done.

As for why it does this, I don't know-- I can confirm that the `_` is there in the names of all the actual C functions in the `nm` output, even on Mac, but I don't know why it does that.

Comment (markusicu) [reply icu4c/source/tools/genren/genren.pl:160]: Thanks for adding the comments, but please swap them :-)

Comment (richgillam) [reply icu4c/source/tools/genren/genren.pl:162]: Dang it!  I caught that mistake, but then forgot to do a "git add".  Hopefully fixed now, and I also went ahead and (optimistically?) squashed my commits.

**Review by markusicu — APPROVED** [review]
lgtm tnx!

**Pull request merged**